aiopg/sa/engine.py

```diff
@@ -2,20 +2,53 @@
 import json
 
 import aiopg
+from sqlalchemy import ColumnDefault
+
 from .connection import SAConnection
 from .exc import InvalidRequestError
-from ..utils import PY_35, _PoolContextManager, _PoolAcquireContextManager
 from ..connection import TIMEOUT
-
+from ..utils import PY_35, _PoolContextManager, _PoolAcquireContextManager
 
 try:
     from sqlalchemy.dialects.postgresql.psycopg2 import PGDialect_psycopg2
+    from sqlalchemy.dialects.postgresql.psycopg2 import PGCompiler_psycopg2
 except ImportError:  # pragma: no cover
     raise ImportError('aiopg.sa requires sqlalchemy')
 
 
+class APGCompiler_psycopg2(PGCompiler_psycopg2):
+    @asyncio.coroutine
+    def aconstruct_params(self, cursor, params=None,
+                          _group_number=None, _check=True):
+        pd = self.construct_params(params, _group_number, _check)
+
+        default_fields = ((f.key, f.default) for f in self.insert_prefetch
+                          if isinstance(f.default, ColumnDefault))
+
+        for key, default in default_fields:
+            pd[key] = (yield from self._exec_default(cursor, default))
+
+        return pd
+
+    @asyncio.coroutine
+    def _exec_default(self, cursor, default):
+        if default.is_sequence:
+            sql = "select nextval('{}')".format(
+                self.dialect.identifier_preparer.format_sequence(default))
```

Comment:
sqlalchemy makes a queries to the db.
Of course, we can make queries in advance...
such behavior we should document it explicitly